lookout/style/format/feature_extractor.py

```diff
@@ -728,21 +717,30 @@ def _parse_file(self, contents: str, root: bblfsh.Node, path: str) -> \
             pos = node.end_position.offset
         return result, parents
 
-    def _remove_labels_with_low_support(self, vnodes: Sequence[VirtualNode]) -> None:
+    def _compute_labels_mappings(self, vnodes: Iterable[VirtualNode]) -> None:
         """
-        Calculate the support for each label and discard those with too little value.
+        Calculate the label to class sequence and class sequence to label mappings.
+
+        Takes into account self.cutoff_label_support and discard those with too little value.
 
         :param vnodes: The virtual nodes extracted from all the files.
         """
-        label_support = defaultdict(int)
+        assert len(self.class_sequences_to_labels) == 0, self.class_sequences_to_labels
+        assert len(self.labels_to_class_sequences) == 0, self.labels_to_class_sequences
+        support = defaultdict(int)
         for vnode in vnodes:
             if vnode.y is not None:
-                label_support[vnode.y] += 1
-        unsupported = {k for k, v in label_support.items() if v < self.cutoff_label_support}
-        for vnode in vnodes:
-            if vnode.y in unsupported:
-                vnode.y = None
-        self._log.debug("Removed %d/%d labels by support %d", len(unsupported), len(label_support),
+                support[vnode.y] += 1
+
+        for class_seq in sorted(support, key=lambda l: -support[l]):
```

Comment:
```python
self.labels_to_class_sequences = [key for key, val in support.items() if val >= self.cutoff_label_support]
```

Reply:
@vmarkovtsev: as discussed above sorting the labels by support makes sense to identify quickly if the label at hand is common or not